### DISCUSSION TITLE
NIP52 + Spring WebSocketClient

### DISCUSSION THREAD

**Discussion by avlo** [tcheeric/nostr-java#179] [General]
@tcheeric hi, eric. if of interest/usefulness to you, the [following PR](https://github.com/avlo/nostr-java-avlo-fork/pull/2) (submitted from-and-to my fork of nostr-java) contains a brief overview and alternative nostr client (Spring WebSocketClient atop Reactive Streams) reference implementation.

below video-capture showing execution of all nostr-java tests using spring WebSocketClient (console, left) and nostr-relay events received (console, right) with subsequent "OK" responses sent by relay and registered (asynchronous awaits()) by client.

https://github.com/user-attachments/assets/59e4cfc7-a918-4e30-83b7-d5982bc239ac

**Comment by tcheeric** [comment]
Wow, this looks awesome. I'll take a detour look at it this week. 

Thanks
-- 
 Sent with Tuta; enjoy secure & ad-free emails: 
 https://tuta.com


10 Sept 2024, 22:35 by ***@***.***:

**Reply by tcheeric** [reply]
Nick, this looks great. I always wanted to accommodate relay responses but my attempts  were always convoluted. I'm still familiarising myself with your pr, and your solution looks very elegant. Thanks.

**Reply by avlo** [reply]
spring is doing all the heavy lifting in this implementation.  a smaller & tighter refactoring (removing manual "sink" management, letting reactive/netty handle them implicitly) will be pushed shortly.  it's further easier to understand and will have a cleaner/more-intuitive client API.  I'll keep you posted.
